src/wallet/rpcwallet.cpp

```diff
@@ -40,7 +40,7 @@
 static const std::string WALLET_ENDPOINT_BASE = "/wallet/";
 static const std::string HELP_REQUIRING_PASSPHRASE{"\nRequires wallet passphrase to be set with walletpassphrase call if wallet is encrypted.\n"};
 
-static inline bool GetAvoidReuseFlag(const CWallet* const pwallet, const UniValue& param) {
+static inline bool GetAvoidReuseFlag(const std::shared_ptr<CWallet> pwallet, const UniValue& param) {
```

Comment:
read-only access to a wallet that is not a `nullptr` should probably be 
```suggestion
static inline bool GetAvoidReuseFlag(const CWallet& wallet, const UniValue& param) {
```

All RPCs check wallet for nullptr at the beginning of the function, so there shouldn't be an occasion where this is assumption fails.

Reply:
Yes, have changed it.
Have also changed the pointer variable name in `ParseIncludeWatchonly`.